core/src/hash_table/sys_entry.rs

```diff
@@ -0,0 +1,194 @@
+// Dna Entry
+//-------------------------------------------------------------------------------------------------
+
+impl ToEntry for Dna {
+    fn to_entry(&self) -> Entry {
+        // TODO #239 - Convert Dna to Entry by following DnaEntry schema and not the to_json() dump
+        Entry::new(EntryType::Dna.as_str(), &self.to_json())
+    }
+
+    fn new_from_entry(entry: &Entry) -> Self {
+        assert!(EntryType::from_str(&entry.entry_type()).unwrap() == EntryType::Dna);
+        return Dna::new_from_json(&entry.content()).expect("entry is not a valid Dna Entry");
+    }
+}
+
+//-------------------------------------------------------------------------------------------------
+// Agent Entry
+//-------------------------------------------------------------------------------------------------
+
+impl ToEntry for Agent {
+    fn to_entry(&self) -> Entry {
+        Entry::new(EntryType::AgentId.as_str(), &self.to_string())
+    }
+
+    fn new_from_entry(entry: &Entry) -> Self {
+        assert!(EntryType::from_str(&entry.entry_type()).unwrap() == EntryType::AgentId);
+        let id_content: String =
+            serde_json::from_str(&entry.content()).expect("entry is not a valid AgentId Entry");
+        Agent::new(Identity::new(id_content))
+    }
+}
+
+//-------------------------------------------------------------------------------------------------
+// UNIT TESTS
+//-------------------------------------------------------------------------------------------------
+
+#[cfg(test)]
+pub mod tests {
+    extern crate test_utils;
+
+    use action::{Action, ActionWrapper};
+    use hash_table::sys_entry::{EntryType, ToEntry};
+    use std::str::FromStr;
+
+    use instance::{tests::test_context, Instance};
+
+    // Committing a DnaEntry to source chain should work
+    #[test]
+    fn can_commit_dna() {
+        // Create Context, Agent, Dna, and Commit AgentIdEntry Action
+        let context = test_context("alex");
+        let dna = test_utils::create_test_dna_with_wat("test_zome", "test_cap", None);
+        let dna_entry = dna.to_entry();
+        let commit_action = ActionWrapper::new(Action::Commit(dna_entry));
+
+        // Set up instance and dispatch action
+        let mut instance = Instance::new();
+        instance.start_action_loop(context);
+        instance.dispatch_and_wait(commit_action);
+
+        // Check if AgentIdEntry is found
+        let mut count = 0;
+        instance
+            .state()
+            .history
+            .iter()
+            .find(|aw| match aw.action() {
+                Action::Commit(entry) => {
+                    assert_eq!(
+                        EntryType::from_str(&entry.entry_type()).unwrap(),
+                        EntryType::Dna
+                    );
+                    count += 1;
+                    true
+                }
+                _ => false,
+            });
+        assert_eq!(count, 1);
```

Comment:
Sonce you aren't using the iterator afterwards, you clould just do a `.count()` on it.

Reply:
done